plugins/native/module/vtkF3DSPZReader.cxx

```diff
@@ -0,0 +1,386 @@
+    pointData->AddArray(sh1Array[i]);
+  }
+
+  SphericalHarmonics<Degree>* sh = begin;
+
+  for (int splatIndex = 0; splatIndex < nbSplats; splatIndex++, sh++)
+  {
+    sh1Array[0]->SetTypedTuple(splatIndex, sh->template decode<-1, 1>().data());
+    sh1Array[1]->SetTypedTuple(splatIndex, sh->template decode<0, 1>().data());
+    sh1Array[2]->SetTypedTuple(splatIndex, sh->template decode<1, 1>().data());
+  }
+
+  if constexpr (Degree >= 2)
+  {
+    vtkNew<vtkUnsignedCharArray> sh2Array[5];
+
+    for (int i = 0; i < 5; i++)
+    {
+      sh2Array[i]->SetNumberOfComponents(3);
+      sh2Array[i]->SetNumberOfTuples(nbSplats);
+      sh2Array[i]->SetName((std::string("sh2") + getSuffix(i - 2)).data());
+      pointData->AddArray(sh2Array[i]);
+    }
+
+    sh = begin;
+
+    for (int splatIndex = 0; splatIndex < nbSplats; splatIndex++, sh++)
+    {
+      sh2Array[0]->SetTypedTuple(splatIndex, sh->template decode<-2, 2>().data());
+      sh2Array[1]->SetTypedTuple(splatIndex, sh->template decode<-1, 2>().data());
+      sh2Array[2]->SetTypedTuple(splatIndex, sh->template decode<0, 2>().data());
+      sh2Array[3]->SetTypedTuple(splatIndex, sh->template decode<1, 2>().data());
+      sh2Array[4]->SetTypedTuple(splatIndex, sh->template decode<2, 2>().data());
+    }
+  }
+
+  if constexpr (Degree >= 3)
+  {
+    vtkNew<vtkUnsignedCharArray> sh3Array[7];
+
+    for (int i = 0; i < 7; i++)
+    {
+      sh3Array[i]->SetNumberOfComponents(3);
+      sh3Array[i]->SetNumberOfTuples(nbSplats);
+      sh3Array[i]->SetName((std::string("sh3") + getSuffix(i - 3)).data());
+      pointData->AddArray(sh3Array[i]);
+    }
+
+    sh = begin;
+
+    for (int splatIndex = 0; splatIndex < nbSplats; splatIndex++, sh++)
+    {
+      sh3Array[0]->SetTypedTuple(splatIndex, sh->template decode<-3, 3>().data());
+      sh3Array[1]->SetTypedTuple(splatIndex, sh->template decode<-2, 3>().data());
+      sh3Array[2]->SetTypedTuple(splatIndex, sh->template decode<-1, 3>().data());
+      sh3Array[3]->SetTypedTuple(splatIndex, sh->template decode<0, 3>().data());
+      sh3Array[4]->SetTypedTuple(splatIndex, sh->template decode<1, 3>().data());
+      sh3Array[5]->SetTypedTuple(splatIndex, sh->template decode<2, 3>().data());
+      sh3Array[6]->SetTypedTuple(splatIndex, sh->template decode<3, 3>().data());
+    }
+  }
+}
+}
+
+//----------------------------------------------------------------------------
+vtkStandardNewMacro(vtkF3DSPZReader);
+
+//----------------------------------------------------------------------------
+int vtkF3DSPZReader::RequestData(
+  vtkInformation*, vtkInformationVector**, vtkInformationVector* outputVector)
+{
+  vtkPolyData* output = vtkPolyData::GetData(outputVector);
+
+  vtkNew<vtkPoints> points;
+  points->SetDataTypeToFloat();
+  output->SetPoints(points);
+
+  vtkSmartPointer<vtkResourceStream> stream;
+
+#if VTK_VERSION_NUMBER > VTK_VERSION_CHECK(9, 4, 20250501)
+  if (this->Stream)
+  {
+    stream = this->Stream;
```

Comment:
easy to reach with a dedicated test I think

Reply:
done